src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java

```diff
@@ -0,0 +1,89 @@
+import java.util.Date;
+import java.util.Set;
+
+@Component
+public class CSVFileParser {
+    private static final int TIMESTAMP_MULTIPLIER = 1000;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ReviewRepository reviewRepository;
+
+    public void parseCsvFile(String filePath) {
+        CSVReader reader = null;
+        try {
+            reader = new CSVReader(new FileReader(filePath));
+            reader.readNext();
+            String[] line;
+            while ((line = reader.readNext()) != null) {
+                if (userRepository.existsById(line[2])) {
+                    User userToUpdate = userRepository.findById(line[2]).get();
+                    userRepository.delete(userToUpdate);
+                    Set<Review> userReviews = userToUpdate.getUserReviews();
+                    Review userReview = getReviewFromReader(line);
+                    reviewRepository.save(userReview);
+                    System.out.println(userReview.getReviewId() + " has been added to the db");
+                    userReviews.add(userReview);
+                    userToUpdate.setUserReviews(userReviews);
+                    userRepository.save(userToUpdate);
+                    System.out.println(userToUpdate.getUserId() + " has been updated in the db");
+                } else {
+                    User newUser = getUserFromReader(line);
+                    Review userReview = getReviewFromReader(line);
+                    reviewRepository.save(userReview);
+                    System.out.println(userReview.getReviewId() + " has been added to the db");
+                    Set<Review> userReviews = Set.of(userReview);
+                    newUser.setUserReviews(userReviews);
+                    userRepository.save(newUser);
+                    System.out.println(newUser.getUserId() + " has been added to the db");
+                }
+            }
+        } catch (CsvValidationException | IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private User getUserFromReader(String[] line) {
+        User user = new User();
+        user.setUserId(line[2]);
+        user.setProfileName(line[3]);
+        user.setProfilePassword(line[3]);
+        user.setUserRoles(Set.of(Role.roleOf("USER")));
+        return user;
+    }
+
+    private Review getReviewFromReader(String[] line) {
+        Review review = new Review();
+        review.setReviewId(Long.parseLong(line[0]));
+        review.setProductId(line[1]);
+        review.setHelpNum(Long.parseLong(line[4]));
+        review.setHelpDenum(Long.parseLong(line[5]));
+        review.setScore(Long.parseLong(line[6]));
+        review.setTime(getTimeFromTimestamp(line[7]));
+        review.setReviewSummary(line[8]);
+        review.setText(line[9]);
+        return review;
+    }
+
+    private String getTimeFromTimestamp(String time) {
+        Timestamp timestamp = new Timestamp(Long.parseLong(time)*TIMESTAMP_MULTIPLIER);
```

Comment:
So, in the end did you manage to get the real time out of it? Could you please send the screenshot of an example in comment below.

Reply:
![image](https://user-images.githubusercontent.com/48065353/87027698-3a1f7380-c1e6-11ea-8174-3c9f30e5f97a.png)

Everything works :)

Reply:
Btw, I tested that with a random timestamp from the .csv file with reviews